src/libraries/System.Runtime/tests/System/CharTests.cs

```diff
@@ -264,6 +308,58 @@ public static void IsDigit_String_Int_Invalid()
             AssertExtensions.Throws<ArgumentOutOfRangeException>("index", () => char.IsDigit("abc", 3)); // Index >= string.Length
         }
 
+        [Fact]
+        public static void IsAsciiHexDigit_Char() =>
+            IsAsciiHexDigit(new HashSet<char>() { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'a', 'b', 'c', 'd', 'e', 'f' });
+
+        [Fact]
+        public static void IsAsciiHexDigitLower_Char() =>
+            IsAsciiHexDigit(new HashSet<char>() { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f' });
+
+        [Fact]
+        public static void IsAsciiHexDigitUpper_Char() =>
+            IsAsciiHexDigit(new HashSet<char>() { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F' });
+
+        private static void IsAsciiHexDigit(HashSet<char> hexDigits)
+        {
+            for (int i = 0; i < 128; i++)
+            {
+                Assert.Equal(hexDigits.Contains((char)i), char.IsAsciiHexDigit((char)i));
+            }
+            foreach (char c in hexDigits)
+            {
+                Assert.False(char.IsAsciiHexDigit((char)(c | 0xFF00)));
+            }
+        }
+
+        [Theory]
+        [InlineData('a', 'a', 'a', true)]
+        [InlineData((char)('a' - 1), 'a', 'a', false)]
+        [InlineData((char)('a' + 1), 'a', 'a', false)]
+        [InlineData('a', 'a', 'b', true)]
+        [InlineData('b', 'a', 'b', true)]
+        [InlineData((char)('a' - 1), 'a', 'b', false)]
+        [InlineData((char)('b' + 1), 'a', 'b', false)]
+        [InlineData('a', 'a', 'z', true)]
+        [InlineData('m', 'a', 'z', true)]
+        [InlineData('z', 'a', 'z', true)]
+        [InlineData((char)('a' - 1), 'a', 'z', false)]
+        [InlineData((char)('z' + 1), 'a', 'z', false)]
+        [InlineData('\0', '\0', '\uFFFF', true)]
+        [InlineData('\u1234', '\0', '\uFFFF', true)]
+        [InlineData('\uFFFF', '\0', '\uFFFF', true)]
+        [InlineData('\u1234', '\u0123', '\u2345', true)]
+        [InlineData('\u1234', '\u2345', '\uFFFF', false)]
+        [InlineData('\u1234', '\u0123', '\u1233', false)]
+        [InlineData('\u1234', '\u0123', '\u1234', true)]
+        [InlineData('\u1234', '\u1235', '\u1231', false)]
```

Comment:
Suggest removing this test case and the test case two lines below. We shouldn't guarantee any specific behavior when the min \& max inputs are bad, since it locks us in to potentially unwanted behavior forevermore.

Reply:
I'll update the XML comments to remove the guarantee, but I'll leave the test case; even if we don't guarantee the behavior, we should still be notified if we accidentally change it, as I expect someone somewhere will take a dependency on it.